app/controllers/app_controller.rb

```diff
@@ -1,24 +1,38 @@
 class AppController < ApplicationController
+  include Mangler
   def main
     render :main, layout: false
   end
 
   def test
     @offers = Offer.all.map {|o| o.format }
     @sessions = Session.all.map {|s| s.json }
-    puts @sessions
     render :test, layout: false
   end
 
   def decision
+    show_decision_view(params)
```

Comment:
renders view when student follows mangled URL, hence must decrypt and find the corresponding offer.